Source/Plugins/Core/com.equella.core/js/tsrc/modules/SearchModule.ts

```diff
@@ -202,3 +220,100 @@ export const searchItems = ({
   };
   return OEQ.Search.search(API_BASE_URL, searchParams);
 };
+
+/**
+ * A function that takes query string params from a shared searching.do URL and converts all applicable params to Search options
+ * @param queryString query string params from a shared `searching.do` URL
+ * @return SearchOptions object, or undefined if there were no query string parameters.
+ */
+export const convertParamsToSearchOptions = async (
+  queryString: string
+): Promise<SearchOptions | undefined> => {
+  if (!queryString) return undefined;
+
+  const params = new URLSearchParams(queryString);
+
+  const getQueryParam = (paramName: LegacyParams) => {
+    return params.get(paramName) ?? undefined;
+  };
+
+  const collectionId = getQueryParam("in")?.substring(1);
+  const ownerId = getQueryParam("owner");
+  const dateRange = getQueryParam("dr");
+  const datePrimary = getQueryParam("dp");
+  const dateSecondary = getQueryParam("ds");
+
+  const getUserDetails = async (
+    userId: string
+  ): Promise<OEQ.UserQuery.UserDetails | undefined> => {
+    const userDetails = await resolveUsers([userId]);
+    return userDetails.length > 0 ? userDetails[0] : defaultSearchOptions.owner;
+  };
+
+  const getCollectionDetails = async (
+    collectionUuid: string
+  ): Promise<Collection[] | undefined> => {
+    const collectionList = await collectionListSummary([
+      OEQ.Acl.ACL_SEARCH_COLLECTION,
+    ]);
+    const filteredCollectionList = collectionList.filter((c) => {
+      return c.uuid === collectionUuid;
+    });
+    return filteredCollectionList.length > 0
+      ? filteredCollectionList
+      : defaultSearchOptions.collections;
+  };
+
+  const RangeType = Union(
+    Literal("between"),
+    Literal("after"),
+    Literal("before"),
+    Literal("on")
+  );
+
+  type RangeType = Static<typeof RangeType>;
+
+  const getLastModifiedDateRange = (
+    rangeType: RangeType,
+    primaryDate?: Date,
+    secondaryDate?: Date
+  ): DateRange | undefined => {
+    if (!primaryDate && !secondaryDate) {
+      return undefined;
+    }
+    return match(
+      [
+        Literal("between"),
+        (): DateRange => ({ start: primaryDate, end: secondaryDate }),
+      ],
+      [Literal("after"), (): DateRange => ({ start: primaryDate })],
+      [Literal("before"), (): DateRange => ({ end: primaryDate })],
+      [
+        Literal("on"),
+        (): DateRange => ({
+          start: primaryDate,
+          end: primaryDate,
+        }),
+      ]
+    )(rangeType.toLowerCase() as RangeType);
```

Comment:
according to the typing `rangeType` should already be `RangeType`?


```suggestion
    )(rangeType.toLowerCase());
```

also since the value comes from a dropdown,

```suggestion
    )(rangeType);
```

may work?